cogni-libs/src/main/java/zone/cogni/libs/sparqlservice/impl/FusekiSparqlService.java

```diff
@@ -39,59 +37,43 @@ public FusekiSparqlService(Config config) {
-      String authEncoded = Base64.getEncoder().encodeToString((config.getUser() + ":" + config.getPassword()).getBytes(StandardCharsets.UTF_8));
-      authHeader = new BasicHeader("Authorization", "Basic " + authEncoded);
-      authenticationHttpContext = createAuthenticationHttpContext(config.getUser(), config.getPassword());
-    }
-    else {
-      authHeader = null; //we're allowed to send null value to the "setHeader" method (whop whop)
-      authenticationHttpContext = null;
-    }
+    httpClient = HttpClientUtils.createHttpClientBuilder(config.getUser(), config.getPassword()).build();
+    queryExecutionBuilder = QueryExecutionHTTPBuilder.service(config.getQueryUrl()).httpClient(httpClient);
   }
 
-
   @Override
   public void uploadTtlFile(File file) {
+    final String sparqlUrl = config.getGraphStoreUrl() + "?graph=" + StringUtils.removeEnd(file.getName(), ".ttl");
+    final HttpRequest request;
     try {
-      String sparqlUrl = config.getGraphStoreUrl() + "?graph=" + StringUtils.removeEnd(file.getName(), ".ttl");
-
-      Response response = Request.Post(sparqlUrl)
-              .setHeader("Content-Type", config.getTurtleMimeType() + ";charset=utf-8")
-              .setHeader(authHeader)
-              .bodyFile(file, ContentType.create(config.getTurtleMimeType(), StandardCharsets.UTF_8))
-              .execute();
-      checkAndDiscardResponse(response);
-    }
-    catch (IOException e) {
+      request = HttpRequest
+          .newBuilder(URI.create(sparqlUrl))
+          .POST(BodyPublishers.ofFile(file.toPath()))
+          .header(CONTENT_TYPE, config.getTurtleMimeType() + ";charset=utf-8")
+          .build();
+    } catch (FileNotFoundException e) {
       throw new RuntimeException(e);
     }
+    execute(request, httpClient);
   }
 
   @Override
   public Model queryForModel(String query) {
-    //pass default graph, method without default graph doesn't use the HttpContext !!!
-    try (QueryExecution queryExecution = QueryExecutionFactory.sparqlService(config.getQueryUrl(), query, null, null, authenticationHttpContext)) {
+    try (QueryExecution queryExecution = queryExecutionBuilder.query(query).build()) {
       return queryExecution.execConstruct();
     }
   }
 
   @Override
   public void executeUpdateQuery(String updateQuery) {
-    try {
-      Response response = Request
-              .Post(config.getUpdateUrl())
-              .setHeader(authHeader)
-              .body(new UrlEncodedFormEntity(Collections.singletonList(new BasicNameValuePair("update", updateQuery)), StandardCharsets.UTF_8))
-              .execute();
-      checkAndDiscardResponse(response);
-    }
-    catch (IOException e) {
-      throw new RuntimeException(e);
-    }
+      final HttpRequest request = HttpRequest
+          .newBuilder(URI.create(config.getUpdateUrl()))
+          .POST(BodyPublishers.ofString("update="+ updateQuery, StandardCharsets.UTF_8))
```

Comment:
is this ok? i think form body is of format key=encodedValue
here the = will also be encoded

Reply:
This code actually does not encode anything. Yet, the [Fuseki test was passing even before](https://github.com/cognizone/asquare/blob/beb6b9bdedbd1c30e932b3ab90ea3fd88b16fd74/cogni-libs/src/test/java/zone/cogni/libs/sparqlservice/impl/AbstractSparqlServiceTest.java#L98), without any encoding in place.

Anyway, I updated the code for percent-encoding the POST params for all service implementations.

Reply:
This code actually does not encode anything. Yet, the [Fuseki test was passing even before](https://github.com/cognizone/asquare/blob/beb6b9bdedbd1c30e932b3ab90ea3fd88b16fd74/cogni-libs/src/test/java/zone/cogni/libs/sparqlservice/impl/AbstractSparqlServiceTest.java#L98), without any encoding in place.

Anyway, I updated the code for percent-encoding the POST params for all service implementations.